### PR TITLE
Layout export quality jpg

### LINKED ISSUE #41559: jpg quality slider in Composer
**Feature description.**
A way to set the jpg compression level when exporting images from the composer.

**Additional context**
Currently the jpeg quality is quite low and shows compression artifacts. I sometimes have to first export in png and then convert to jpg with a 95% quality from imagemagick.
A slider that would be in one of the export dialogs would remove that step for me. It could be either in the "Save Layout As" (file browsing) dialog or the following "Export Options" dialog.
It could also show up in the "Atlas" tab of the composer under "Output".

### PR DESCRIPTION
**Pull request by YoannQDQ**
- Fix #41559

## Description

Add a `quality` parameter in `QgsLayoutExporter::ImageExportSettings`.

For jpeg and jpg, the layout image export dialog now displays an additional line (hidden for other extensions)

image_quality

### CODE REVIEW

**Comment by YoannQDQ** [comment]
Not sure how to handle the spell check fail, as `exportLayersAsSeperateFiles` is part of the public API (and I didn't modify it in this PR)

File: src/core/layout/qgslayoutexporter.h [hunk 1]
```diff
@@ -231,6 +235,14 @@ class CORE_EXPORT QgsLayoutExporter
        */
       QVector<qreal> predefinedMapScales;
 
+
+      /**
+       * Image quality, typically used for JPEG compression (whose quality ranges from 0 to 100)
```
Comment (rouault) [inline src/core/layout/qgslayoutexporter.h:240]: I'm not sure which formats would accept 0. For JPEG the minimum is 1

Comment (YoannQDQ) [reply src/core/layout/qgslayoutexporter.h]: Not sure either. I set the minimum to 1.

File: src/app/layout/qgslayoutdesignerdialog.h [hunk 2]
```diff
@@ -563,8 +563,8 @@ class QgsLayoutDesignerDialog: public QMainWindow, public Ui::QgsLayoutDesignerB
     void showRasterizationWarning();
     void showForceVectorWarning();
 
-    bool showFileSizeWarning();
-    bool getRasterExportSettings( QgsLayoutExporter::ImageExportSettings &settings, QSize &imageSize );
+    bool showFileSizeWarning()   ;
```
Comment (nyalldawson) [inline src/app/layout/qgslayoutdesignerdialog.h:566]:
```suggestion
    bool showFileSizeWarning();
```

File: src/app/layout/qgslayoutdesignerdialog.h [hunk 3]
```diff
@@ -563,8 +563,8 @@ class QgsLayoutDesignerDialog: public QMainWindow, public Ui::QgsLayoutDesignerB
...
+    bool getRasterExportSettings( QgsLayoutExporter::ImageExportSettings &settings, QSize &imageSize, const QString &fileExtension = QString() );
```
Comment (nyalldawson) [inline src/app/layout/qgslayoutdesignerdialog.h:567]: We can safely modify the signature here, as this method isn't exposed to python:

```suggestion
    bool getRasterExportSettings( QgsLayoutExporter::ImageExportSettings &settings, QSize &imageSize, const QString &fileExtension );
```

File: src/gui/layout/qgslayoutimageexportoptionsdialog.cpp [hunk 4]
```diff
@@ -23,9 +23,11 @@
 
 #include <QCheckBox>
 #include <QPushButton>
+#include <QSpinBox>
+#include <QSlider>
 
-QgsLayoutImageExportOptionsDialog::QgsLayoutImageExportOptionsDialog( QWidget *parent, Qt::WindowFlags flags )
-  : QDialog( parent, flags )
+QgsLayoutImageExportOptionsDialog::QgsLayoutImageExportOptionsDialog( QWidget *parent, const QString &fileExtension, Qt::WindowFlags flags )
+  : QDialog( parent, flags ), mFileExtension( fileExtension )
```
Comment (nyalldawson) [inline src/gui/layout/qgslayoutimageexportoptionsdialog.cpp:30]:
```suggestion
  : QDialog( parent, flags )
  , mFileExtension( fileExtension )
```

File: src/gui/layout/qgslayoutimageexportoptionsdialog.cpp [hunk 5]
```diff
@@ -34,6 +36,16 @@ QgsLayoutImageExportOptionsDialog::QgsLayoutImageExportOptionsDialog( QWidget *p
...
   connect( mClipToContentGroupBox, &QGroupBox::toggled, this, &QgsLayoutImageExportOptionsDialog::clipToContentsToggled );
   connect( mHelpButtonBox, &QDialogButtonBox::helpRequested, this, &QgsLayoutImageExportOptionsDialog::showHelp );
+
+  const bool showQuality = shouldShowQuality();
+  mQualitySpinBox->setVisible( showQuality );
+  mQualitySlider->setVisible( showQuality );
+  mQualityLabel->setVisible( showQuality );
+  mQualityLabel->setText( tr( "%1 quality", "Image format" ).arg( mFileExtension.toUpper() ) );
+
+  connect( mQualitySpinBox, static_cast < void ( QSpinBox::* )( int ) > ( &QSpinBox::valueChanged ), mQualitySlider, &QSlider::setValue );
```
Comment (nyalldawson) [inline src/gui/layout/qgslayoutimageexportoptionsdialog.cpp:46]:
```suggestion
  connect( mQualitySpinBox, qOverload< int >( &QSpinBox::valueChanged ), mQualitySlider, &QSlider::setValue );
```

**Review by nyalldawson — APPROVED** [review]

**Pull request merged**